results/GTK.md

```diff
@@ -0,0 +1,147 @@
+
+
+- **Building SWT binaries using GTK4**
+SWT binaries can also be build using GTK4 on windows OS.
+Changes are committed to https://github.com/swt-initiative31/prototype-gtk/commit/0bbc7708db718fd749bf000e3328424376dcbfe4
+
+
+- **Embedding of other window handles (AWT)**
+It is possible to generate swt-awt-gtk-4966r2.dll on Windows, which allows to run SWT snippets 154, 155, 300, and 361 that use the SWT.EMBEDDED flag.
+The approach followed includes creating a Windows-specific swt_awt_win32.c file, and the respective object file is used in make_win32.mak to build the SWT-AWT binaries.
+However, further verification is needed to determine which scenarios fail.
+ I noticed that maximizing the window causes the contents to disappear for  300 and Snippet 361 runs but fails on further interaction.
+The changes are committed to https://github.com/swt-initiative31/prototype-gtk/commit/e5414eb4d83ae109fb2fa3651784d5d234e9cbed
+
+
+- **Edge browser integration (On going)**
+Because GTK does not support the webkitGTK port on Windows OS and if we need to embed the Edge browser, we would have to use Microsoft OLE, which was previously used for SWT with Win32. 
+The approach followed will be a platform dependent browser which still uses win32 APIs to build SWT binaries for browser support.
+Current approach followed is to reuse the SWT's browser codebase which uses Microsoft OLE and windows libraries and also win32 APIs.
+Using this approach able to build SWT binaries having both GTK and Win32 APIs building together . 
+Snippets related browser are failing at runtime currently.
+-- Not able to receive the pointer to Webview2 Controller from OLE framework which uses asynchronous call back mechanisms ) and analysis is still going on.
+-- Able to receive the pointer to webview2 Environment which uses asynchronous call back mechanisms 
+
+
+### Achievements
+
+<!-- What did you achieve in terms of concrete prototyping work? That may include code, documentation and other artifacts. What features/functionality can you show with these artifacts? Which milestones did you reach and what functionality do they cover and demonstrate? It would also be great to see some screenshot in specifically this section (which may be outsourced to the Appendix to ease readability of this section). -->
+
+ - Able to decide Multiple Display instances creation is no more supported in SWT on Windows OS if we use GTK3/GTK4
+ - Able to build SWT binaries using GTK4 on windows OS which runs almost fine. Each and every snippet is not validated yet.
+ - Able to build SWT binaries for AWT component to work in SWT (Able to embed AWT into SWT)
+
+### Insights
+
+<!-- What insights did you gain when developing the prototype? What were the central obstacles that might also influence a full SWT implementation based on that technology? What was complicated to achieve? What did take long? -->
+- **Yet Unsupported Features(Not worked on):**
+-- PrintDialog
```

Comment:
Printing support on windows SWT using GTK is not just matter of integration because we cannot use LINUX SWT code of printer as it is in Windows.
The reasons being GTK have enhanced support for printing on Linux using CUPS(Common UNIX Printing System) . GTK have separate folder called 'unix-print' which contains APIs usable only on Linux.

SWT on windows using GTK is having limited support w.r.t printer capabilities and cannot use APIs under 'unix-print' .

As per this [comment](https://bugs.eclipse.org/bugs/show_bug.cgi?id=488431#c46) .

GTK offers the callback-based model for printing on both Linux and Windows. while SWT uses a job-based system. This is a core part of GTK's printing API, regardless of the platform. However, the difference lies in the additional support GTK provides for job-based printing on Unix/Linux platforms, which is absent or less mature on Windows.
(This has to be analyzed yet if it still applies !)